### PR TITLE
Refactor get latest analysis

### PR DESCRIPTION
**Pull request by henrikstranneheim**
## Description

Refactor `get_latest_analysis` and migrate to CRUD.

### Added

- Tests

### Changed

- Refactor `get_latest_analysis` and migrate to CRUD.
- Refactor tests
- Rename case_id fixtures to case_name

### How to prepare for test
- [ ] ssh to hasta (depending on type of change)
- [ ] activate stage: `us`
- [ ] request trailblazer-stage on hasta: `paxa`
- [ ] install on stage:
    ```Shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_trailblazer -t trailblazer -b refactor_get_latest_analysis -a
    ```
- [ ] ssh to clinical-db (depending on type of change)
- [ ] install on stage:
`bash /home/proj/production/servers/resources/clinical-db.scilifelab.se/update-trailblazer-ui-stage.sh [THIS-BRANCH-NAME]`

### How to test
- [ ] login to ...
- [ ] do ...

### Expected test outcome
- [ ] check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: tests/cli/test_cli_core.py [hunk 1]
```diff
@@ -173,46 +221,67 @@ def test_delete_nonexisting(cli_runner, trailblazer_context, caplog):
...
 
-        # WHEN trying to delete ongoing analysis without --force flag
-        result = cli_runner.invoke(delete, [str(analysis_obj.id)], obj=trailblazer_context)
+    # WHEN trying to delete ongoing analysis without --force flag
+    result = cli_runner.invoke(delete, [str(analysis.id)], obj=trailblazer_context)
 
-        # THEN command executes successfully
-        assert result.exit_code == 0
+    # THEN command executes successfully
+    assert result.exit_code == process_exit_success
```
Comment (islean) [inline tests/cli/test_cli_core.py:247]: Feels a bit weird to me that we consider this a success. Probably not within the scope of this PR even if it should be changed, though.

File: tests/cli/test_cli_core.py [hunk 2]
```diff
@@ -173,46 +221,67 @@ def test_delete_nonexisting(cli_runner, trailblazer_context, caplog):
...
+    # WHEN running command with --force flag
+    result = cli_runner.invoke(delete, [str(analysis.id), "--force"], obj=trailblazer_context)
 
-        # THEN log informs user that Trailblazer is deleting analysis
-        assert "Deleting" in caplog.text
+    # THEN command runs successfully
+    assert result.exit_code == process_exit_success
 
+    # THEN log informs user that Trailblazer is deleting analysis
+    assert "Deleting" in caplog.text
```
Comment (islean) [inline tests/cli/test_cli_core.py:279]: maybe an "assert analysis.status == TrailblazerStatus.CANCELLED"

File: trailblazer/store/api.py [hunk 3]
```diff
@@ -111,29 +108,27 @@ def set_analysis_completed(self, analysis_id: int) -> None:
...
             raise ValueError(f"Invalid status. Allowed values are: {TrailblazerStatus.statuses()}")
-        analysis_obj: Analysis = self.get_latest_analysis(case_id=case_id)
-        analysis_obj.status: str = status
+        analysis: Optional[Analysis] = self.get_latest_analysis_for_case(case_name=case_id)
+        analysis.status = status
         self.commit()
-        LOG.info(f"{analysis_obj.family} - Status set to {status.upper()}")
+        LOG.info(f"{analysis.family} - Status set to {status.upper()}")
 
     def add_comment(self, case_id: str, comment: str):
```
Comment (islean) [inline trailblazer/store/api.py:125]: Probably out of scope for this PR but would we want to match so all "case_id" switch to "case_name"?

**Review by islean — APPROVED** [review]
Looks nice! 🔥

File: tests/cli/test_cli_core.py [hunk 4]
```diff
@@ -36,60 +35,81 @@ def test_base(cli_runner):
...
     result = cli_runner.invoke(
-        set_analysis_status, ["--status", "qc", failed_analysis], obj=trailblazer_context
+        set_analysis_status, ["--status", "qc", failed_analysis_case_name], obj=trailblazer_context
     )
 
     # THEN command runs successfully
-    assert result.exit_code == 0
+    assert result.exit_code == process_exit_success
 
     # THEN status will be set to COMPLETED
```
[the remark this answers is not in the document]
Comment (henrikstranneheim) [reply tests/cli/test_cli_core.py:94]: Fixed

File: tests/cli/test_cli_core.py (hunk 1 above)
Comment (henrikstranneheim) [reply tests/cli/test_cli_core.py:247]: Agreed -we should raise an error, but in another PR

File: tests/cli/test_cli_core.py (hunk 2 above)
Comment (henrikstranneheim) [reply tests/cli/test_cli_core.py:279]: It is gone, but added this:
```
    # THEN the analysis should have been deleted from the database
    assert not trailblazer_db.get_latest_analysis_for_case(case_name=analysis.family)

```

File: trailblazer/store/api.py (hunk 3 above)
Comment (henrikstranneheim) [reply trailblazer/store/api.py:119]: Another PR

**Pull request merged**